src/Farmer/Builders/Builders.ContainerService.fs

```diff
@@ -0,0 +1,127 @@
+[<AutoOpen>]
+module Farmer.Builders.ContainerService
+
+open Farmer
+open Farmer.Arm.ContainerService
+open Farmer.CoreTypes
+open Farmer.Vm
+
+type AgentPoolConfig =
+    { Name : ResourceName
+      Count : int
+      Mode : AgentPoolMode
+      OsDiskSize : int<Gb>
+      OsType : OS
+      VmSize : VMSize
+    }
+
+type AksConfig =
+    { Name : ResourceName
+      AgentPools : AgentPoolConfig list
+      DnsPrefix : string
+      EnableRBAC : bool
+      LinuxProfile : (string * string list) option
+      ServicePrincipalClientID : string option
+      WindowsProfileAdminUserName : string option
+    }
+    interface IBuilder with
+        member this.DependencyName = this.Name
+        member this.BuildResources location = [
+            // VM itself
+            { Name = this.Name
+              Location = location
+              DnsPrefix = this.DnsPrefix
+              EnableRBAC = this.EnableRBAC
+              AgentPoolProfiles = this.AgentPools |> List.map (fun agentPool ->
+                  {| Name = agentPool.Name
+                     Count = agentPool.Count
+                     Mode = agentPool.Mode
+                     OsDiskSize = agentPool.OsDiskSize
+                     OsType = agentPool.OsType
+                     VmSize = agentPool.VmSize |})
+              LinuxProfile =
+                  this.LinuxProfile
+                  |> Option.map (fun (username, keys) -> {| AdminUserName = username; PublicKeys = keys |})
+              ServicePrincipalProfile =
+                  this.ServicePrincipalClientID
+                  |> Option.map (fun clientId -> {| ClientId = clientId; ClientSecret = SecureParameter (sprintf "client-secret-for-%s" this.Name.Value) |})
+              WindowsProfile =
+                  this.WindowsProfileAdminUserName
+                  |> Option.map (fun username -> {| AdminUserName = username; AdminPassword = SecureParameter (sprintf "admin-password-for-%s" this.Name.Value) |})
+            }
+        ]
+
+type AgentPoolBuilder() =
+    member _.Yield _ =
+        { Name = ResourceName.Empty
+          Count = 1
```

Comment:
Nitpicking, but I'd default to 2 nodes - lots of interesting edge cases start popping up inside K8s cluster use as soon as node count > 1